syntax = 'proto3';

package mason.library;

message PortSchema {
  string name = 1;
  string type = 2;
  string direction = 3;
  bool sequence = 4;
  bool map = 5;
  repeated string choices = 6;
  string default = 7;
  string format = 8;
  string visibility = 9;
}

message NodeSchema {
  string group = 1;
  string name = 2;
  string shape = 3;
  string default_label = 4;
  repeated PortSchema ports = 5;
  repeated string signals = 6;
  repeated string slots = 7;
}

message BlueprintSchema {
  string group = 1;
  string name = 2;
  repeated string signals = 4;
  repeated string slots = 5;
}

message Library {
  repeated NodeSchema nodes = 1;
  repeated BlueprintSchema blueprints = 2;
}
